### PR TITLE
Pydantic V2 migration

### PR DESCRIPTION
**Pull request by hamishfagg**
There are a few more things changed between pydantic V1 and V2. I've run through this migration guide: https://docs.pydantic.dev/latest/migration/

### CODE REVIEW

File: mindsdb/integrations/handlers/mysql_handler/settings.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class ConnectionConfig(BaseModel):
     password: str = None
     database: str = None
 
-    @root_validator(pre=True)
+    @model_validator(pre=True)
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/mysql_handler/settings.py:14]: should be `@model_validator(mode="before")`

File: mindsdb/integrations/handlers/twelve_labs_handler/settings.py [hunk 2]
```diff
@@ -80,9 +80,9 @@ class TwelveLabsHandlerModel(BaseModel):
     prompt: Optional[str] = None
 
     class Config:
-        extra = Extra.forbid
+        extra = "forbid"
 
-    @root_validator(pre=True, allow_reuse=True, skip_on_failure=True)
+    @model_validator(pre=True, skip_on_failure=True)
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/twelve_labs_handler/settings.py:85]: @model_validator(mode="before")

File: mindsdb/integrations/handlers/twelve_labs_handler/settings.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ def check_param_typos(cls, values):
 
         return values
 
-    @root_validator(pre=True, allow_reuse=True, skip_on_failure=True)
+    @model_validator(pre=True, skip_on_failure=True)
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/twelve_labs_handler/settings.py:105]: @model_validator(mode="before")

File: mindsdb/integrations/handlers/twelve_labs_handler/settings.py [hunk 4]
```diff
@@ -127,7 +127,7 @@ def check_for_valid_task(cls, values):
...
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/twelve_labs_handler/settings.py:130]: @model_validator(mode="before")

**Comment by dusvyat** [comment]
no 'pre' argument in `model_validator`, its been replaced with `mode='before'` see [here](https://docs.pydantic.dev/latest/concepts/validators/#validation-of-default-values)

**Comment by hamishfagg** [comment]
@dusvyat fixed those issues, can you take another look? Can test on https://dev.mindsdb.com too

**Review by dusvyat — APPROVED** [review]
lgtm

**Pull request merged**